cluster-autoscaler/provisioningrequest/service/v1beta1client/client.go

```diff
@@ -0,0 +1,145 @@
+		return nil, err
+	}
+
+	ptLister, err := newPodTemplatesLister(kubeConfig, make(chan struct{}))
+	if err != nil {
+		klog.Warningf("Failed to create Pod Template lister, will not use Provisioning Requests. Error was: %v", err)
+		return nil, err
+	}
+
+	return &ProvisioningRequestClient{
+		client:   prClient,
+		prLister: prLister,
+		ptLister: ptLister,
+	}, nil
+}
+
+// ProvisioningRequest gets a specific ProvisioningRequest CR.
+func (c *ProvisioningRequestClient) ProvisioningRequest(namespace, name string) (*v1beta1.ProvisioningRequest, error) {
+	return c.prLister.ProvisioningRequests(namespace).Get(name)
+}
+
+// ProvisioningRequests gets all ProvisioningRequest CRs.
+func (c *ProvisioningRequestClient) ProvisioningRequests() ([]*v1beta1.ProvisioningRequest, error) {
+	provisioningRequests, err := c.prLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("error fetching provisioningRequests: %w", err)
+	}
+	return provisioningRequests, nil
+}
+
+// FetchPodTemplates fetches PodTemplates referenced by the Provisioning Request.
+func (c *ProvisioningRequestClient) FetchPodTemplates(pr *v1beta1.ProvisioningRequest) ([]*apiv1.PodTemplate, error) {
+	podTemplates := make([]*apiv1.PodTemplate, 0, len(pr.Spec.PodSets))
+	for _, podSpec := range pr.Spec.PodSets {
+		podTemplate, err := c.ptLister.PodTemplates(pr.Namespace).Get(podSpec.PodTemplateRef.Name)
+		if errors.IsNotFound(err) {
+			klog.Infof("While fetching Pod Template for Provisioning Request %s/%s received not found error", pr.Namespace, pr.Name)
+			continue
+		} else if err != nil {
+			return nil, err
+		}
+		podTemplates = append(podTemplates, podTemplate)
+	}
+	return podTemplates, nil
+}
+
+// newPRClient creates a new Provisioning Request client from the given config.
+func newPRClient(kubeConfig *rest.Config) (*versioned.Clientset, error) {
+	return versioned.NewForConfig(kubeConfig)
+}
+
+// newAllPRsLister creates a lister for the Provisioning Requests in the cluster.
+func newAllPRsLister(prClient versioned.Interface, stopChannel <-chan struct{}) (listers.ProvisioningRequestLister, error) {
+	factory := externalversions.NewSharedInformerFactory(prClient, 1*time.Hour)
+	prLister := factory.Autoscaling().V1beta1().ProvisioningRequests().Lister()
+	factory.Start(stopChannel)
+	informersSynced := factory.WaitForCacheSync(stopChannel)
+	for _, synced := range informersSynced {
+		if !synced {
+			return nil, fmt.Errorf("can't create Provisioning Request lister")
+		}
+	}
+	klog.V(2).Info("Successful initial Provisioning Request sync")
+	return prLister, nil
+}
+
+// newPodTemplatesLister creates a lister for the Pod Templates in the cluster.
+func newPodTemplatesLister(kubeConfig *rest.Config, stopChannel <-chan struct{}) (v1.PodTemplateLister, error) {
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		klog.Warningf("Failed to create Pod Template lister, will not use Provisioning Requests Error was: %v", err)
```

Comment:
This will be logged twice, as the error will be again handled in `NewProvisioningRequestClient`. Maybe it would make sense to create the client directly in `NewProvisioningRequestClient`, similarly how it is done for PR lister?